qiime2/sdk/action.py

```diff
@@ -14,30 +14,62 @@
 
 import decorator
 import dill
+import parsl
+from parsl.app.app import python_app, join_app
 
 import qiime2.sdk
 import qiime2.core.type as qtype
 import qiime2.core.archive as archive
 from qiime2.core.util import LateBindingAttribute, DropFirstParameter, tuplize
+from qiime2.sdk.parsl_config import PARSL_CONFIG, setup_parsl
+from qiime2.sdk.proxy import Proxy
 
 
-def _subprocess_apply(action, ctx, args, kwargs):
+def _subprocess_apply(action, ctx, execution_ctx, args, kwargs):
```

Comment:
I don't think we need to accept `execution_ctx`, because `foo.asynchronous()` is the only thing that calls this helper, so it's fixed as `{type: asynchronous}` for the purposes of `action._bind()`